monkey/common/di_container.py

```diff
@@ -0,0 +1,122 @@
+import inspect
+from typing import Any, MutableMapping, Sequence, Type, TypeVar
+
+T = TypeVar("T")
+
+
+class UnregisteredTypeError(ValueError):
+    pass
+
+
+class DIContainer:
```

Comment:
Have you looked into this: https://python-dependency-injector.ets-labs.org/examples/application-single-container.html ?